src/System.Buffers.Experimental/System/Buffers/OwnedArray.cs

```diff
@@ -27,13 +27,10 @@ public OwnedArray(T[] array)
 
         public override int Length => _array.Length;
```

Comment:
Still think forcing override of `Memory` by making it abstract in `MemoryManager<T>` and dropping the `Length` property would be better; so this would become:
```diff
-   public override int Length => _array.Length;
+   public override Memory<T> Memory => new Memory<T>(this, 0, _array.Length);
```
but ¯\\_(ツ)_/¯

Reply:
Fun fact: the original proposal did not have `MemoryManager<T>` implementing `IMemoryOwner<T>` at all (because the two types have two different responsibilities), so if you both subclassed `MemoryManager<T>` _and_ implemented `IMemoryOwner<T>` you'd be required to implement the method manually anyway. Once the decision was made to make the base class implement the interface then it turned into "If you can infer a reasonable behavior, make it virtual. Only make it abstract if you can't infer a reasonable behavior." And that's how the API shape ended up the way it is now.

(the_more_you_know.jpg)

Reply:
Does the virtual length property add anything over (or have utility over): `manager.Memory.Length`?

Its default implementation is `manager.GetSpan().Length` which is more costly; with then the default `Memory` becoming the super costly `new Memory<T>(this, 0, this.GetSpan().Length);` and its two properties for the implementor to get out of sync (e.g. `.Length` and `Memory.Length` returning different things)

> Only make it abstract if you can't infer a reasonable behavior."
 
If you want it `virtual` go with the dirty (as it would currently evaluate to):
```csharp
public virtual Memory<T> Memory => new Memory<T>(this, 0, GetSpan().Length);
```

Just not sure what the `.Length` property adds?

Reply:
Literally the only reason it remains is to minimize code churn from `OwnedMemory<T>`.

One could perhaps make an argument that a subclass could implement the `Length` property getter more cheaply than the `GetSpan()` method, which makes the `Memory<T>(MemoryManager<T>, ...)` ctor's validation a bit simpler. But since the `GetSpan()` method is supposed to be fast + straightforward anyway I don't think this is a huge benefit.

Reply:
Regardless, this PR is not the place to debate the design of this type. If you have a concrete suggestion for an improvement along with justification for it then open an issue in the _corefx_ repo.

Reply:
>  If you have a concrete suggestion for an improvement along with justification for it then open an issue in the corefx repo.

https://github.com/dotnet/corefx/issues/28751